src/cluster_legacy.c

```diff
@@ -3613,13 +3625,17 @@ void clusterLinkConnectHandler(connection *conn) {
          * replaced by the clusterSendPing() call. */
         node->ping_sent = old_ping_sent;
     }
-    /* We can clear the flag after the first packet is sent.
-     * If we'll never receive a PONG, we'll never send new packets
-     * to this node. Instead after the PONG is received and we
-     * are no longer in meet/handshake status, we want to send
-     * normal PING packets. */
-    node->flags &= ~CLUSTER_NODE_MEET;
-
+    /* NOTE: We cannot clear the MEET flag from the node until we get a response
+     * from the other node. If the MEET packet is not accepted by the other node
```

Comment:
I think I understand the comments from the discussion but ignoring the discussion for a second and pretending this is the first time I am reading this code comment, I find that the references to "the node", "this node", "the other node", a bit confusing. I would suggest describing the scenario using some logical node IDs like "node A is recently created and is asked to `CLUSTER MEET` node B, so on and so forth".

Reply:
The note reads better with Node A/Node B convention. Thanks!